File: sol/739.cpp
struct Solution {
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n = temperatures.size(), idx;
        vector<int> ans(n, 0), st;
        for (int i = 0; i < n; ++i){
            while (!st.empty() && temperatures[i] > temperatures[st.back()]){
                idx = st.back();
                st.pop_back();
                ans[idx] = i - idx;
            }
            st.push_back(i);
        }
        return ans;
    }
};

File: sol/739.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n = len(temperatures)
        ans = [0] * n
        st = []
        for i, t in enumerate(temperatures):
            while len(st) != 0 and t > temperatures[st[-1]]:
                idx = st.pop()
                ans[idx] = i - idx
            st.append(i)
        return ans
